libstuff/SHTTPSManager.cpp

```diff
@@ -80,107 +62,78 @@ int SStandaloneHTTPSManager::getHTTPResponseCode(const string& methodLine) {
+    uint64_t elapsed = now - timeoutFromTime;
+    int size = transaction.fullResponse.deserialize(transaction.s->recvBuffer);
+    bool specificallyTimedOut = transaction.timeoutAt < now;
+    if (size) {
+        // Consume how much we read.
+        transaction.s->recvBuffer.consumeFront(size);
+
+        // 200OK or any content?
+        transaction.finished = now;
+        if (SContains(transaction.fullResponse.methodLine, " 200 ") || transaction.fullResponse.content.size()) {
+            // Pass the transaction down to the subclass.
+            if (_onRecv(&transaction)) {
+                // If true, then the transaction was closed in onRecv.
+                return;
             }
+            SASSERT(transaction.response);
         } else {
-            // Haven't timed out yet, let the caller know how long until we do.
-            nextActivity = min(nextActivity, timeoutFromTime + timeout);
+            // Error, failed to authenticate or receive a valid server response.
+            SWARN("Message failed: '" << transaction.fullResponse.methodLine << "'");
+            transaction.response = 500;
         }
-
-        // If we're done, remove from the active and add to completed
-        if (active->response) {
-            // Switch lists
-            SINFO("Completed request '" << active->fullRequest.methodLine << "' to '" << active->fullRequest["Host"]
-                  << "' with response '" << active->response << "' in '" << elapsed / 1000 << "'ms");
-            _activeTransactionList.erase(activeIt);
-            _completedTransactionList.push_back(active);
-            completedRequests.push_back(active);
+    } else if (transaction.s->state.load() > Socket::CONNECTED || elapsed > timeout || specificallyTimedOut) {
+        // Net problem. Did this transaction end in an inconsistent state?
+        SWARN("Connection " << (elapsed > timeout ? "timed out" : "died prematurely") << " after " << elapsed / 1000 << "ms");
+        transaction.response = transaction.s->sendBufferEmpty() ? 501 : 500;
+        if (transaction.response == 501) {
+            SHMMM("SStandaloneHTTPSManager: '" << transaction.fullRequest.methodLine
+                  << "' timed out receiving response in " << (elapsed / 1000) << "ms.");
         }
+    } else {
+        // Haven't timed out yet, let the caller know how long until we do.
+        nextActivity = min(nextActivity, timeoutFromTime + timeout);
+    }
+
+    // If we're done, remove from the active and add to completed
+    if (transaction.response) {
+        // Switch lists
+        SINFO("Completed request '" << transaction.fullRequest.methodLine << "' to '" << transaction.fullRequest["Host"]
+              << "' with response '" << transaction.response << "' in '" << elapsed / 1000 << "'ms");
     }
 }
 
 SStandaloneHTTPSManager::Transaction::Transaction(SStandaloneHTTPSManager& manager_) :
     s(nullptr),
     created(STimeNow()),
     finished(0),
+    timeoutAt(0),
```

Comment:
This is the replacement for all of the complicated lists in BedrockServer to track what's timed out or not. We jsut attach the timeout to the transaction.